src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -14075,6 +14075,9 @@ void Unit::ProcDamageAndSpellFor(bool isVictim, Unit* target, uint32 procFlag, u
     // Player is loaded now - do not allow passive spell casts to proc
     if (GetTypeId() == TYPEID_PLAYER && ToPlayer()->GetSession()->PlayerLoading())
         return;
+    // Player is dead - do not proc on dead player
```

Comment:
So if "this" is not necessarily a player, we need to check for `GetTypeId() == TYPEID_PLAYER` in addition to `!IsAlive()` --- is that the preferred way to do it?